subscriptions_test.go

```diff
@@ -759,28 +759,92 @@ func TestSubscriptions_Create_TransactionError(t *testing.T) {
+			    <reference>3383506</reference>
+			    <source>subscription</source>
+			    <recurring type="boolean">false</recurring>
+			    <test type="boolean">true</test>
+			    <voidable type="boolean">false</voidable>
+			    <refundable type="boolean">false</refundable>
+			    <ip_address>127.0.0.1</ip_address>
+			    <transaction_error>
+			      <error_code>declined_saveable</error_code>
+			      <error_category>soft</error_category>
+			      <merchant_message>The transaction was declined without specific information.  Please contact your payment gateway for more details or ask the customer to contact their bank.</merchant_message>
+			      <customer_message>The transaction was declined. Please use a different card or contact your bank.</customer_message>
+			      <gateway_error_code nil="nil"/>
+			    </transaction_error>
+			    <cvv_result code="" nil="nil"/>
+			    <avs_result code="" nil="nil"/>
+			    <avs_result_street nil="nil"/>
+			    <avs_result_postal nil="nil"/>
+			    <created_at type="datetime">2017-03-15T21:21:16Z</created_at>
+			    <updated_at type="datetime">2017-03-15T21:21:16Z</updated_at>
+			    <details>
+			      <account>
+			        <account_code>1</account_code>
+			        <first_name>Verena</first_name>
+			        <last_name>Example</last_name>
+			        <company>New Company Name</company>
+			        <email>verena@example.com</email>
+			        <billing_info type="credit_card">
+			          <first_name>Verena</first_name>
+			          <last_name>Example</last_name>
+			          <address1>123 Main St.</address1>
+			          <address2 nil="nil"/>
+			          <city>San Francisco</city>
+			          <state>CA</state>
+			          <zip>94105</zip>
+			          <country>US</country>
+			          <phone nil="nil"/>
+			          <vat_number nil="nil"/>
+			          <card_type>Visa</card_type>
+			          <year type="integer">2019</year>
+			          <month type="integer">12</month>
+			          <first_six>400000</first_six>
+			          <last_four>0341</last_four>
+			        </billing_info>
+			      </account>
+			    </details>
 			  </transaction>
 			</errors>`)
 	})
 
-	r, _, err := client.Subscriptions.Create(recurly.NewSubscription{})
+	r, newSubscription, err := client.Subscriptions.Create(recurly.NewSubscription{})
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	} else if !r.IsError() {
 		t.Fatal("expected create subscription to return OK")
-	} else if r.Transaction.UUID != "3d1c6aa86e3d447eb0f3b4a6e3e074d9" {
-		t.Fatalf("unexpected transaction uuid: %s", r.Transaction.UUID)
-	} else if r.TransactionError.ErrorCode != "fraud_security_code" {
-		t.Fatalf("unexpected transaction error code: %s", r.TransactionError.ErrorCode)
+	} else if newSubscription.Transaction == nil {
+		t.Fatal("expected transaction to be set")
+	} else if !reflect.DeepEqual(newSubscription.Transaction.TransactionError, &recurly.TransactionError{
```

Comment:
Let's `reflect.DeepEqual` against the entire `Transaction` -- not just the `TransactionError` embedded in the `Transaction`.